Story/StoryMenu.cs

```diff
@@ -483,5 +502,207 @@ public string GetCampaignName(SlugcatStats.Name name)
+                configurableBools.Add(MoreSlugcats.MMF.cfgNoRandomCycles);
+                configurableBools.Add(MoreSlugcats.MMF.cfgOldTongue);
+                configurableBools.Add(MoreSlugcats.MMF.cfgQuieterGates);
+                configurableBools.Add(MoreSlugcats.MMF.cfgSafeCentipedes);
+                configurableBools.Add(MoreSlugcats.MMF.cfgSandboxItemStems);
+                configurableBools.Add(MoreSlugcats.MMF.cfgScavengerKillSquadDelay);
+                configurableBools.Add(MoreSlugcats.MMF.cfgShowUnderwaterShortcuts);
+                configurableBools.Add(MoreSlugcats.MMF.cfgSpeedrunTimer);
+                configurableBools.Add(MoreSlugcats.MMF.cfgSurvivorPassageNotRequired);
+                configurableBools.Add(MoreSlugcats.MMF.cfgSwimBreathLeniency);
+                configurableBools.Add(MoreSlugcats.MMF.cfgThreatMusicPulse);
+                configurableBools.Add(MoreSlugcats.MMF.cfgTickTock);
+                configurableBools.Add(MoreSlugcats.MMF.cfgUpwardsSpearThrow);
+                configurableBools.Add(MoreSlugcats.MMF.cfgVanillaExploits);
+                configurableBools.Add(MoreSlugcats.MMF.cfgVulnerableJellyfish);
+                configurableBools.Add(MoreSlugcats.MMF.cfgWallpounce);
+
+                configurableFloats.Add(MoreSlugcats.MMF.cfgRainTimeMultiplier);
+                configurableFloats.Add(MoreSlugcats.MMF.cfgSlowTimeFactor);
+
+                configurableInts.Add(MoreSlugcats.MMF.cfgHunterBonusCycles);
+                configurableInts.Add(MoreSlugcats.MMF.cfgHunterCycles);
+
+                for (int i = 0; i < hostBoolRemixSettings.Count; i++)
+                {
+                    configurableBools[i]._typedValue = hostBoolRemixSettings[i];
+                    RainMeadow.Debug($"CLIENT GETTING BOOL KEY {configurableBools[i].key}, VALUE: {configurableBools[i]._typedValue}");
+
+                }
+
+                for (int i = 0; i < hostFloatRemixSettings.Count; i++)
+                {
+                    configurableFloats[i]._typedValue = hostFloatRemixSettings[i];
+                    RainMeadow.Debug($"CLIENT GETTING FLOAT KEY {configurableFloats[i].key}, VALUE: {configurableFloats[i]._typedValue}");
+
+                }
+
+
+                for (int i = 0; i < hostIntRemixSettings.Count; i++)
+                {
+                    configurableInts[i]._typedValue = hostIntRemixSettings[i];
+                    RainMeadow.Debug($"CLIENT GETTING INT KEY {configurableInts[i].key}, VALUE: {configurableInts[i]._typedValue}");
+
+                }
+
+            }
+        }
+
+        internal List<bool> GetHostBoolStoryRemixSettings()
+        {
+            List<Configurable<bool>> configurableBools = new List<Configurable<bool>>();
+
+            List<bool> configurableTypes = new List<bool>();
+
+            if (ModManager.MMF)
+            {
+
+
+                configurableBools.Add(MoreSlugcats.MMF.cfgAlphaRedLizards);
```

Comment:
You can use reflection to iterate through these fields rather than hard-code them. It'd be easier to read and understand

Reply:
[187d855](https://github.com/henpemaz/Rain-Meadow/pull/114/commits/187d855a6af7d37ac95f995019971784c91da7df) implemented reflection.

Tried both making one broad function that [returned a tuple ](https://github.com/henpemaz/Rain-Meadow/pull/114/commits/187d855a6af7d37ac95f995019971784c91da7df#diff-36e272de8c1f4642c955628b5b462c7f8abdb920eceeac001d0904d58575db47R157) and by writing three individual functions.

However, it seems that going the reflection route makes the serializer very unhappy and fails to serialize the data:

```
[Error  :RainMeadow] 17:57:31|977|Serializer.ICustomSerializable.GetSerializationMethod:System.Collections.Generic.List1[System.Boolean] not handled by SerializerCallMethod
[Error  :RainMeadow] 17:57:31|977|Serializer.ICustomSerializable.GetSerializationMethod:System.Collections.Generic.List1[System.Single] not handled by SerializerCallMethod
[Error  :RainMeadow] 17:57:31|977|Serializer.ICustomSerializable.GetSerializationMethod:System.Collections.Generic.List`1[System.Int32] not handled by SerializerCallMethod
```